aws/rust-runtime/aws-sigv4/src/http_request.rs

```diff
@@ -10,10 +10,9 @@
 //! **Note**: This requires `http0-compat` to be enabled.
 //!
 //! ```rust
-//! # use aws_credential_types::Credentials;
+//! use aws_credential_types::Credentials;
 //! use aws_smithy_runtime_api::client::identity::Identity;
-//! # use aws_sigv4::http_request::SignableBody;
-//! #[cfg(feature = "http0-compat")]
```

Comment:
as the docs state above, this example requires the `http0-compat` feature to be enabled (that's why the example is gated on this feature)

Reply:
Maybe I am missing something. Per my understanding, it requires the http0-compat feature to be enabled in Cargo.toml

`aws-sigv4 = { version = "1.1.1", default-features = false, features = ["http0-compat", "sign-http", "sign-eventstream", "sigv4a", "num-bigint"] }`

However, when I copy the example code into my client app,  #[cfg(feature = "http0-compat")] code will check whether http0-compact feature is enabled in my app (vs. on aws-sigv4 crate)

Reply:
oh I see what you're saying! Yeah the fix here is actually to add an additional `#` at the beginning of those lines to suppress them from docs.rs

We should suppress the `fn` test line too

The example itself needs the config gating so that it will compile even when this feature isn't enabled.

Reply:
@rcoh Ah... I got your point. This example needs to be build as part of rustdocs and it should not be built when the feature is not enabled. I will update PR